pkg/cache/v3/linear.go

```diff
@@ -326,19 +361,25 @@ func (cache *LinearCache) computeResponse(watch watch, replyEvenIfEmpty bool) (W
 
 		// changedResources is already filtered based on the subscription.
 		resourcesToReturn = changedResources
-	case sub.IsWildcard():
+	case sub.IsWildcard() && !cache.customWildcardMode:
 		// Include all resources for the type.
 		resourcesToReturn = make([]string, 0, len(cache.resources))
 		for resourceName := range cache.resources {
 			resourcesToReturn = append(resourcesToReturn, resourceName)
 		}
 	default:
+		combinedResources := maps.Clone(sub.SubscribedResources())
```

Comment:
Same remarks. This should not clone if not needed and below is making all requests wildcards